psi4/driver/driver_cbs.py

```diff
@@ -1497,11 +1228,577 @@ def _cbs_gufunc(func, total_method_name, **kwargs):
+                                 metadata[0]["wfn"], metadata[0]["options"])
+    GRAND_NEED.append(
+        dict(
+            zip(d_fields, [
+                'scf', metadata[0]["scheme"],
+                _contract_bracketed_basis(metadata[0]["basis"][0]), metadata[0]["wfn"], metadata[0]["alpha"], NEED, +1,
+                0.0, None, None, None, None
+            ])))
+    if len(metadata) > 1:
+        for delta in metadata[1:]:
+            NEED = _expand_scheme_orders(delta["scheme"], delta["basis"][0], delta["basis"][1], delta["wfn"],
+                                         delta["options"])
+            GRAND_NEED.append(
+                dict(
+                    zip(d_fields, [
+                        delta["stage"], delta["scheme"],
+                        _contract_bracketed_basis(delta["basis"][0]), delta["wfn"], delta["alpha"], NEED, +1, 0.0,
+                        None, None, None, None
+                    ])))
+            NEED = _expand_scheme_orders(delta["scheme"], delta["basis_lo"][0], delta["basis_lo"][1], delta["wfn_lo"],
+                                         delta["options_lo"])
+            GRAND_NEED.append(
+                dict(
+                    zip(d_fields, [
+                        delta["stage"], delta["scheme"],
+                        _contract_bracketed_basis(delta["basis_lo"][0]), delta["wfn_lo"], delta["alpha"], NEED, -1,
+                        0.0, None, None, None, None
+                    ])))
+
+    # MODELCHEM is unordered, possibly redundant list of single result *entries* needed to satisfy full CBS
+    # JOBS is subset of MODELCHEM with minimal list of single result *jobs* needed to satisfy full CBS
+    # TROVE is superset of JOBS with maximal list of single result *entries* resulting from JOBS
+    # "entry" here is a mtd-bas-opt spec that can support E/G/H data
+    # "job" here is an entry on which to sic Psi4 that, through VARH, may fill in multiple entries
+
+    MODELCHEM = []
+    for stage in GRAND_NEED:
+        for lvl in stage['d_need'].values():
+            MODELCHEM.append(lvl)
+
+    # Apply chemical reasoning to choose the minimum computations to run
+    JOBS = MODELCHEM[:]
+    listfmt = """   {:>12} / {:24} for  {}{}\n"""
+
+    #     Remove duplicate modelchem portion listings
+    for mc in MODELCHEM:
+        dups = -1
+        for indx_job, job in enumerate(JOBS):
+            #print('')
+            #print(job['f_wfn'], mc['f_wfn'], job['f_wfn'] == mc['f_wfn'])
+            #print(job['f_basis'], mc['f_basis'], job['f_basis'] == mc['f_basis'])
+            #print(job['f_options'], mc['f_options'], job['f_options'] == mc['f_options'],
+            #      job['f_options'] is not False)
+            if ((job['f_wfn'] == mc['f_wfn']) and (job['f_basis'] == mc['f_basis'])
+                    and (job['f_options'] == mc['f_options'])):  # and (job['f_options'] is not False)):
+                # TODO: I don't understand why I need this change
```

Comment:
What change, exactly?

Reply:
the commented bit on 1306. I've moved the todo aside with explanation. same in next loop